### PR TITLE
2.x: rename Observable and Single #doOnCancel to #doOnDispose

### PR DESCRIPTION
**Pull request by Mauin**
#4456

Completable already used .doOnDispose() however I renamed the arguments to match Observable and Single

### CODE REVIEW

File: src/main/java/io/reactivex/Single.java [hunk 1]
```diff
@@ -1579,13 +1579,13 @@
...
      * </dl>
-     * @param onCancel the runnable called when the subscription is cancelled (disposed)
+     * @param onDispose the runnable called when the subscription is disposed
      * @return the new Single instance
      * @since 2.0
      */
-    public final Single<T> doOnCancel(final Action onCancel) {
-        ObjectHelper.requireNonNull(onCancel, "onCancel is null");
-        return RxJavaPlugins.onAssembly(new SingleDoOnCancel<T>(this, onCancel));
+    public final Single<T> doOnDispose(final Action onDispose) {
```
Comment (vanniktech) [inline src/main/java/io/reactivex/Single.java:1586]: should not we deprecate `doOnCancel` and then remove it with RC3?

**Comment by akarnokd** [comment]
The diff for observable doesn't show but the offline-comparison only lists 30-40 lines changes. Did you run some whitespace cleanup?

File: src/main/java/io/reactivex/Single.java (hunk 1 above)
Comment (akarnokd) [reply src/main/java/io/reactivex/Single.java:1586]: I think the discussion #4451 came to a consensus that no need for going soft but delete what's no longer needed before releasing RC2,  hence no deprecation.

**Comment by Mauin** [comment]
@akarnokd Whoops, you're right. Sorry. Auto formatter seems to have changed some whitespaces. I'll revert that.

File: src/main/java/io/reactivex/Single.java (hunk 1 above)
Comment (vanniktech) [reply src/main/java/io/reactivex/Single.java:1586]: Alright works for me too

**Comment by akarnokd** [comment]
:+1:

**Comment by pwittchen** [comment]
This update is missing in release notes on GitHub. Please, remember to mention such changes in the future.

**Pull request merged**